src/Sql/SecretsManager/dbo/Stored Procedures/ApiKey/ApiKey_Create.sql

```diff
@@ -13,12 +14,19 @@ AS
 BEGIN
     SET NOCOUNT ON
 
+    IF (@HashedClientSecret IS NULL)
+    BEGIN
+      DECLARE @hb VARBINARY(128) = HASHBYTES('SHA2_256', @ClientSecret);
+      SET @HashedClientSecret = CAST(N'' as xml).value('xs:base64Binary(sql:variable("@hb"))', 'varchar(128)');
+    END
```

Comment:
We technically don't need this since the data migration script will take care of it, and the duration this would be active is quite short.

Reply:
I wrote this based on the [rollback documentation](https://contributing.bitwarden.com/contributing/database-migrations/edd#rollbacks). If our rollback process redeploys the old application code, and we don't have this, we will need to run the migration again for any newly created tokens.

Reply:
That's true!